gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/crud_service/access_point/AccessPointCrudServiceImpl.java

```diff
@@ -60,34 +61,52 @@ public void updateAccessPoints(
                 .build();
             var existingAccessPoints = accessPointRepository.findByCriteria(accessPointCriteria, null, null);
 
+            var apsToCreate = new ArrayList<io.gravitee.repository.management.model.AccessPoint>();
+            var apsToDelete = new ArrayList<>(existingAccessPoints);
+
             for (AccessPoint accessPoint : accessPoints) {
                 var newAccessPoint = AccessPointAdapter.INSTANCE.fromEntity(accessPoint);
 
-                var existingAP = existingAccessPoints
+                var existingAPsForTarget = existingAccessPoints
                     .stream()
                     .filter(existingAccessPoint -> existingAccessPoint.getTarget().equals(newAccessPoint.getTarget()))
-                    .findFirst();
+                    .toList();
 
-                // create the received access point if it doesn't exist
-                if (existingAP.isEmpty()) {
-                    createAccessPoint(newAccessPoint);
+                // if there aren't access points for this target, it needs to be added for creation
+                if (existingAPsForTarget.isEmpty()) {
+                    apsToCreate.add(newAccessPoint);
                     continue;
                 }
 
-                // verify if there is an access point for this target and if it was modified
-                var modifiedExistingAP = existingAP.filter(existingAccessPoint ->
-                    !existingAccessPoint.getHost().equals(newAccessPoint.getHost()) ||
-                    !existingAccessPoint.isSecured() == newAccessPoint.isSecured() ||
-                    !existingAccessPoint.isOverriding() == newAccessPoint.isOverriding()
-                );
-
-                // and if that's the case, then a new AP needs to be created and the existing one needs to be mark as deleted
-                if (modifiedExistingAP.isPresent()) {
-                    var apToDelete = modifiedExistingAP.get();
-                    this.createAccessPoint(newAccessPoint);
-                    this.deleteAccessPoint(apToDelete);
+                // Search in the existing access points if the new one is already present (comparing equality)
+                // if yes => it needs to be removed from the list of APs to delete
+                //          and the new AP doesn't need to be created
+                // if no => then it will remain in the deletion list to be cleaned up
+                //          the new AP doesn't exist and needs to be added for creation
+                var shouldCreateNewAP = true;
+                for (var existingAccessPoint : existingAPsForTarget) {
+                    if (
+                        existingAccessPoint.getHost().equals(newAccessPoint.getHost()) &&
```

Comment:
Shouldn't we rely on EqualsAndHash to avoid any issue in the future?

Reply:
yes you are right. Using just `equals` is not enough for a host. I'll update it